### PR TITLE
Handle HTML double comment opens

### LINKED ISSUE #65: Nested comments aren't parsed correctly
Nested comments aren't handled correctly

    // Source
     -->

    // Expected
     -->

    // Real
    <!--

### PR DESCRIPTION
**Pull request by jnvsor**
Fixes #65 

Strangely enough, while it's more complex than the last commit it actually speeds it up according to a quick check (Probably because it skips the first element, so files without comments will skip the comment parsing altogether)

### CODE REVIEW

File: test/source/comments/double-open.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+layout: compress
+---
+
+1 <!-- <!-- --> -->
```
Comment (penibelst) [inline test/source/comments/double-open.html]: This line is invalid according to the [W3C Validator](https://validator.w3.org/nu/#textarea).

```html
<!DOCTYPE html>
<html>
<head>
<title>Test</title>
</head>
<body>
<p></p>
1 <!-- <!-- --> -->
</body>
</html>
```
> Consecutive hyphens did not terminate a comment. `--` is not permitted inside a comment, but e.g. `- -` is.

Can we use valid cases only here?

**Comment by penibelst** [comment]
I have mixed feelings about this change, since nobody have ever reported a bug with nested comments. But if you say it’s faster now, I don’t care about complexity.

**Comment by jnvsor** [comment]
It might just be faster if the benchmarks are biased towards files without comments, but looking in the performance folder they all seem to have one.

As for it being invalid - last I checked this was still how browsers handled it (Even in xhtml strict mode IIRC) and lines 4/7 are fixed while the validator says they're fine.

Could I ask you to check the benchmarks yourself? It's not enough of a speedup to merit a commit on it's own.

There is a part that fixes a bug if you ever encounter a file that starts with a comment, but it's equally unlikely as lines 4/7

I've stuck a loop around the files in performance to get a larger and hopefully more reliable measurement, let me run a few and get back to you.

**Comment by jnvsor** [comment]
Ok, just ran longer tests on these. It's still faster but I have *no* idea why.

I made a "Faster" version hybrid trying to get a faster time and it ended up taking longer, so here are the numbers:

    This patch
      0.000000   0.000000  56.680000 ( 56.731578)
    Before this patch
      0.000000   0.000000  58.440000 ( 58.498990)
    The "Fast hybrid" version
      0.000000   0.000000  58.970000 ( 59.022384)

So yeah. No idea why but it's faster. (Consider the precedent of throwing in dead or useless code just to make it faster before merging!)

**Comment by penibelst** [comment]
My tests looks similar to yours:

```bash
# before
0.000000   0.010000  13.170000 ( 13.202535)
# after
0.000000   0.000000  13.090000 ( 13.101123)
```

It *is* faster, so I’m in.

File: src/compress.liquid [hunk 2]
```diff
@@ -46,13 +46,27 @@
   {% assign _comments = site.compress_html.comments %}
 {% endif %}
 {% if _comments.size == 2 %}
-  {% assign _comment_befores = _content | split: _comments.first %}
+  {% capture _comment_befores %}.{{ _content }}{% endcapture %}
```
Comment (penibelst) [inline src/compress.liquid]: What does the dot do here?

Comment (penibelst) [inline src/compress.liquid]: Oh, I understand. It’s a placeholder for pages that starts with a comment, right?

Comment (jnvsor) [inline src/compress.liquid]: Handles an edge case where your page starts with a comment: Unlikely, 
(And definitely non-compliant) but if it happens it will mess up the split.

There's another potential edge case if you set 
`site.compress_html.comments.first` to start with a dot but that's 
unlikely in any language never mind html.

On another note - I found a bug in blanklines my tests didn't catch. 
Makes no semantic difference but anyway:

If a line starts with a <pre> the last line before it will be included 
in front of it, so a blank line will end up indenting it, and a line 
with something on it will end up in front of it.

Nothing serious but I'll see what I can do about it.

Comment (jnvsor) [inline src/compress.liquid]: Yep.
